src/transformers/testing_utils.py

```diff
@@ -927,6 +961,18 @@ def require_torch_gpu(test_case):
     return unittest.skipUnless(torch_device == "cuda", "test requires CUDA")(test_case)
 
 
+def require_torch_gpu_if_bnb_not_multi_backend_enabled(test_case):
+    """
+    Decorator marking a test that requires a GPU if bitsandbytes multi-backend feature is not enabled.
+    """
+    if is_bitsandbytes_available():
+        import bitsandbytes as bnb
+
+        if hasattr(bnb, "features") and "multi_backend" in bnb.features:
```

Comment:
There are multiple versions of this idiom around this PR now: this `hasattr(bnb, "features") and "multi_backend" in bnb.features` and `"multi_backend" in getattr(bnb, "features", set())` multiple times elsewhere.

Feels like these should be a single

```python
def is_bitsandbytes_multi_backend_available() -> bool:
    import bitsandbytes as bnb
    return "multi_backend" in getattr(bnb, "features", set())
```